packages/core/src/physics/DynamicCollider.ts

```diff
@@ -2,24 +2,215 @@ import { IDynamicCollider } from "@oasis-engine/design";
+  }
+
+  set mass(newValue: number) {
+    this._mass = newValue;
+    (<IDynamicCollider>this._nativeCollider).setMass(newValue);
+  }
+
+  /**
+   * The center of mass relative to the transform's origin.
+   */
+  get centerOfMass(): Vector3 {
+    return this._centerOfMass;
+  }
+
+  set centerOfMass(newValue: Vector3) {
+    if (this._centerOfMass !== newValue) {
+      newValue.cloneTo(this._centerOfMass);
+    }
+    (<IDynamicCollider>this._nativeCollider).setCenterOfMass(this._centerOfMass);
+  }
+
+  /**
+   * The diagonal inertia tensor of mass relative to the center of mass.
+   */
+  get inertiaTensor(): Vector3 {
+    return this._inertiaTensor;
+  }
+
+  set inertiaTensor(newValue: Vector3) {
+    if (this._inertiaTensor !== newValue) {
+      newValue.cloneTo(this._inertiaTensor);
+    }
+    (<IDynamicCollider>this._nativeCollider).setInertiaTensor(this._inertiaTensor);
+  }
+
+  /**
+   * The maximum angular velocity of the collider measured in radians per second. (Default 7) range { 0, infinity }.
+   */
+  get maxAngularVelocity(): number {
+    return this._maxAngularVelocity;
+  }
+
+  set maxAngularVelocity(newValue: number) {
+    this._maxAngularVelocity = newValue;
+    (<IDynamicCollider>this._nativeCollider).setMaxAngularVelocity(newValue);
+  }
+
+  /**
+   * Maximum velocity of a collider when moving out of penetrating state.
+   */
+  get maxDepenetrationVelocity(): number {
+    return this._maxDepenetrationVelocity;
+  }
+
+  set maxDepenetrationVelocity(newValue: number) {
+    this._maxDepenetrationVelocity = newValue;
+    (<IDynamicCollider>this._nativeCollider).setMaxDepenetrationVelocity(newValue);
+  }
+
+  /**
+   * The mass-normalized energy threshold, below which objects start going to sleep.
+   */
+  get sleepThreshold(): number {
+    return this._sleepThreshold;
+  }
+
+  set sleepThreshold(newValue: number) {
+    this._sleepThreshold = newValue;
+    (<IDynamicCollider>this._nativeCollider).setSleepThreshold(newValue);
+  }
+
+  /**
+   * The solverIterations determines how accurately collider joints and collision contacts are resolved.
+   */
+  get solverIterations(): number {
+    return this._solverIterations;
+  }
+
+  set solverIterations(newValue: number) {
+    this._solverIterations = newValue;
+    (<IDynamicCollider>this._nativeCollider).setSolverIterations(newValue);
+  }
+
+  /**
+   * Controls whether physics affects the dynamic collider.
+   */
+  get isKinematic(): boolean {
+    return this._isKinematic;
+  }
+
+  set isKinematic(newValue: boolean) {
+    this._isKinematic = newValue;
+    (<IDynamicCollider>this._nativeCollider).setIsKinematic(newValue);
+  }
+
+  /**
+   * Controls whether physics will change the rotation of the object.
+   */
+  get freezeRotation(): boolean {
+    return this._freezeRotation;
+  }
+
+  set freezeRotation(newValue: boolean) {
+    this._freezeRotation = newValue;
+    (<IDynamicCollider>this._nativeCollider).setFreezeRotation(newValue);
+  }
```

Comment:
Maybe the function is similar to constraints

Reply:
+1